ex9/js/main.js

```diff
@@ -0,0 +1,37 @@
+requirejs(['ajaxGet','load','search', 'order'] , function(ajaxGet, load, search, order) {
+  $(function() {
+    var fun = {
+      data: {},
+      load: function(){
+         ajaxGet('json/data.json', function(phoneBookData) {
+           fun.data =  phoneBookData;
+          fun.print( fun.data );
+        });
+      },
+      sortByAsc: function() {
+          fun.print( _.sortBy(fun.data, 'name') );
+      },
+      sortByDesc: function() {
+        fun.print( _.sortBy(fun.data, 'name').reverse());
+      },
+      filter: function() {
+          var filterList = _.filter(fun.data, function(name) {
+            if (name.name.indexOf($( '#filter' ).val()) > -1) return name;
```

Comment:
filter expects your function to return true or false (http://underscorejs.org/#filter). You currently return `name` or `undefined` instead.